jetty-server/src/main/java/org/eclipse/jetty/server/session/AbstractSessionCache.java

```diff
@@ -583,8 +637,11 @@ public void release(String id, Session session) throws Exception
                     }
                     else
                     {
-                        //reactivate the session
+                        //reactivate the session, ignoring any (non-serializable)
+                        //attributes set by the activate listener
+                        boolean dirty = session.getSessionData().isDirty();
```

Comment:
you can now remove this dirty logic from here

Reply:
Done.